tools/inspect/preferred_namespace_check.cpp

```diff
@@ -0,0 +1,107 @@
+//  deprecated macro check implementation  ----------------------------------//
+//  Protect against ourself: hpxinspect:nodeprecated_macros
+
+//  Copyright (c) 2015 Brandon Cordes
+//  Based on the depreciated_macro_check by Eric Niebler
+//
+//  Distributed under the Boost Software License, Version 1.0.
+//  (See accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+
+
+#include "preferred_namespace_check.hpp"
+#include <functional>
+#include <vector>
+#include "boost/regex.hpp"
+#include "boost/lexical_cast.hpp"
+#include "boost/filesystem/operations.hpp"
+
+namespace fs = boost::filesystem;
+
+namespace
+{
+    //Add additional unwanted namespaces in dont_want
+    const char * dont_want[] = {
+        "boost::move",
+        "std::begin",
+        "std::end",
+        NULL
+    };
+    //Add replacements for each dont_want in want
+    //both dont_want and want must be the same size
+    const char * want[] = {
+        "std::move",
+        "boost::begin",
+        "boost::end",
+        NULL
+    };
+} // unnamed namespace
```

Comment:
It might be better to have an array of pairs of `char const* const` instead. That would make future additions less error prone.